### DISCUSSION TITLE
How to use proper typing in uploading file?

### DISCUSSION THREAD

**Discussion by donPuerto** [quasarframework/quasar#10437] [General - Components / Directives / etc]
Again  Error on uploading file with Typecript.

How to use proper typing on file?

**Picker quasar component**
![image](https://user-images.githubusercontent.com/9329005/130201919-c011e55d-1c20-4f9c-91e8-d1db22078907.png)
The nice thing with it, you will receive and object rather than the array 

...
While **Uploader** you will received an Array
![image](https://user-images.githubusercontent.com/9329005/130202108-975e4b9f-8e74-4e4a-b0c0-12798d4ff6c4.png)

Now on plain vanilla Javascript you do like this **files.value[0].name**
But on typescript, you will get an error see below

```js
const uploadAvatar  = (event: Event) => {
  const files = ref(event)
  console.log('files', (files.value)[0])   ---------> Errror
}

Error:
Element implicitly has an 'any' type because expression of type '0' can't be used to index type '{ readonly bubbles: boolean; cancelBubble: boolean; readonly cancelable: boolean; readonly composed: boolean; readonly currentTarget: { addEventListener: (type: string, listener: EventListenerOrEventListenerObject | null, options?: boolean | ... 1 more ... | undefined) => void; dispatchEvent: (event: Event) => boole...'.
  Property '0' does not exist on type '{ readonly bubbles: boolean; cancelBubble: boolean; readonly cancelable: boolean; readonly composed: boolean; readonly 
```
Event Log
![image](https://user-images.githubusercontent.com/9329005/130202435-1df36205-af30-4a37-b13e-ec26609ad6cf.png)

**Comment by MatthewSH** [comment]
You'd have to specify the type. It's right to error, you're not telling it what `files` is. 
If it's an array you could do something like: 
```javascript
const files: File[] = ref(event)
```

I haven't tested this with `ref` so it may behave differently and this is all off the top of my head knowledge about TypeScript and Quasar, if you encounter issues, I'll be happy to dig more into it.
